core/src/mindustry/entities/comp/WeaponsComp.java

```diff
@@ -96,10 +96,30 @@ public void update(){
                 mount.rotation = 0;
                 mount.targetRotation = angleTo(mount.aimX, mount.aimY);
             }
-
+            
+            if(mount.shoot && ammo <= 0){
```

Comment:
This is a massive amount of duplicated code just to play a sound. I can't accept this.

Reply:
Working on a possible change to combine the two.